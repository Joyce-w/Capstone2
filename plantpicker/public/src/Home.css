.Home img{
    min-height: 120%;
    min-width:min-content;
    position: fixed;
    width: 100%;
    height: fit-content;
    top: -170px;
    left: 0;
    z-index: -1; /* Image moves to back to act like bg picture*/
}

.Home-div{
    background-color: #ffffff53;
    border-radius: 5%;
    padding: 0 3em;
    position: absolute;
    left: 10%;
    top: 15%;
    height: 47em;
    width: 37em;
}

.Home h1{
    font-family: 'Montserrat', sans-serif;
    font-size: 48px;
}

.Home p{
    /* font-family: 'Raleway',
    sans-serif; */
}