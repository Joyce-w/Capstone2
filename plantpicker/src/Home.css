.Home img{
    min-height: 120%;
    min-width: min-content;
    position: fixed;
    width: 100%;
    height: fit-content;
    top: -170px;
    left: 0;
    z-index: -1;  
    /* Image moves to back to act like bg picture*/
}

/* .Home{
    width: 100vw;
    height: 100vh;
    background: url("https://images.unsplash.com/photo-1572969057162-d3b00f790462?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1955&q=8") no-repeat center center fixed;
    background-size: cover;
      height: 100%;
      overflow: hidden;
} */

.Home-leftDiv{
    width: 45%;    
    height: 100vh;
    position: absolute;
    padding: 3.5% 5%
}

.Home-div{
    position: relative;
    background-color: #ffffff53;
    border-radius: 5%;
    padding: 3em 3em;
    width: 70%;
    height: 70vh;
    margin: auto;
    overflow: auto;
}

.Home h1{
    font-size: 48px;
    margin-top: 0;
}

.Home button{
    position: relative;
    margin:1em;
    width: 11em;
    height: 2.5em;
    background-color: #A7D7C5;
    border: none;
    border-radius: 10px;
    font-size: larger;
    font-weight: bold;

}

.Home button:hover {
    background-color: #84c7aee7;
}


/**MEDIA QUERIES**/
@media screen and (max-width: 1024px) {
    .Home-leftDiv {
        width: 100%;
        padding: none;
    }
}