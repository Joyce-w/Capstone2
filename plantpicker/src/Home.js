import "./Home.css";
import { Link } from 'react-router-dom';


function Home() {

    
    return (
        <div className="Home">
            <img alt="monsterea plant" src="https://images.unsplash.com/photo-1572969057162-d3b00f790462?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1955&q=80"></img>

            <div className="Home-leftDiv">
                <div className="Home-div" aria-label="Plant picture by @domain on Unsplash.com">
                    <h1> Plant + Pot</h1>
                    <p>New empty space you want to fill?</p>
                    <p>Want to gift some greenery but not sure what? </p>
                    <p>Take the quiz to help you get matched with the perfect plant. Create an account to save plants for later or make some lists so you can create that jungle room you always wanted.</p>

                    <br></br>
                    <br></br>
                <Link to="/register"><button className="Home-signup-btn">Signup</button></Link>                    
                <Link to="/quiz"><button className="Home-quiz-btn">Start Quiz</button>  </Link>                  
                </div>

            </div>

                     
        </div>

    )
}

export default Home;