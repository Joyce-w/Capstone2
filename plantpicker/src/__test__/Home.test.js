import { render, fireEvent, screen } from "@testing-library/react";
import {MemoryRouter} from 'react-router-dom'
import Home from "../Home";

it("renders w/o crashing", () => {
    render(<MemoryRouter><Home/></MemoryRouter>)
})

it("matches snapshot", function () {
    const { asFragment } = render(<MemoryRouter><Home /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
})

test("if links are on the page", () => {
    const { getByText } = render(<MemoryRouter><Home /></MemoryRouter>);

    let signupBtn = getByText("Signup");
    let quiz = getByText("Start Quiz");
    let header = getByText("Plant + Pot");

    expect(signupBtn).toBeInTheDocument();
    expect(quiz).toBeInTheDocument();
    expect(header).toBeInTheDocument();
})

test("redirects to correct component", async () => {
    const { getByText, getByLabelText, debug } = render(<MemoryRouter><Home /></MemoryRouter>);

    //toggle button to direct to signup page
    // let signupBtn = getByText("Signup");
    // fireEvent.click(signupBtn);
    // fireEvent.click(getByLabelText(/Signup/))
    // const inputNode = screen.getByLabelText('Email', {selector: 'input'})
    // expect(inputNode).toBeInTheDocument();
})