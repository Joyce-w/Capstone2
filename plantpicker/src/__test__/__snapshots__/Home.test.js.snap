// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <img
      alt="monsterea plant"
      src="https://images.unsplash.com/photo-1572969057162-d3b00f790462?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1955&q=80"
    />
    <div
      class="Home-leftDiv"
    >
      <div
        aria-label="Plant picture by @domain on Unsplash.com"
        class="Home-div"
      >
        <h1>
           Plant + Pot
        </h1>
        <p>
          New empty space you want to fill? Want to gift some greenery? Take the quiz to help you get started. Create an account so you can save plants for later or make build on some list so you can create that jungle room.
        </p>
        <button
          class="Home-signup-btn"
        >
          <a
            href="/register"
          >
            Signup
          </a>
        </button>
        <button
          class="Home-quiz-btn"
        >
          <a
            href="/quiz"
          >
            Start Quiz
          </a>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
